modules/porous_flow/include/materials/PorousFlowSimpleCO2.h

```diff
@@ -0,0 +1,261 @@
+
+  /**
+   * Derivative of the density of gaseous CO2 as a function of
+   * pressure.
+   *
+   * @param pressure fluid pressure (Pa)
+   * @param temperature fluid temperature (C)
+   * @return derivative of CO2 density (kg/m^3) with respect to pressure (Pa)
+   */
+  Real dGasDensity_dP(Real pressure, Real temperature) const;
+
+  /**
+   * Derivative of the density of gaseous CO2 as a function of
+   * temperature.
+   *
+   * @param pressure fluid pressure (Pa)
+   * @param temperature fluid temperature (C)
+   * @return derivative of CO2 density (kg/m^3) with respect to temperature (C)
+   */
+  Real dGasDensity_dT(Real pressure, Real temperature) const;
+
+  /**
+   * Derivative of the density of supercritical CO2 as a function of
+   * pressure.
+   *
+   * @param pressure fluid pressure (Pa)
+   * @param temperature fluid temperature (C)
+   * @return derivative of CO2 density (kg/m^3) with respect to pressure (Pa)
+   */
+  Real dSupercriticalDensity_dP(Real pressure, Real temperature) const;
+
+  /**
+   * Derivative of the density of supercritical CO2 as a function of
+   * temperature.
+   *
+   * @param pressure fluid pressure (Pa)
+   * @param temperature fluid temperature (C)
+   * @return derivative of CO2 density (kg/m^3) with respect to temperature (C)
+   */
+  Real dSupercriticalDensity_dT(Real pressure, Real temperature) const;
+
+  /**
+   * The derivative of CO2 viscosity with respect to density
+   *
+   * @param pressure fluid pressure (Pa)
+   * @param temperature fluid temperature (C)
+   * @return derivative of CO2 viscosity wrt density
+   */
+  Real dViscosity_dDensity(Real pressure, Real temperature, Real density) const;
+
+  /**
+   * Derivative of the viscosity of gaseous CO2 as a function of density
+   *
+   * @param temperature fluid temperature (C)
+   * @param density fluid density (kg/m^3)
+   * @return derivative of CO2 viscosity wrt density
+   */
+  Real dGasViscosity_dDensity(Real temperature, Real density) const;
+
+  /**
+   * Derivative of the viscosity of supercritical CO2 as a function of density
+   *
+   * @param pressure fluid pressure (Pa)
+   * @param temperature fluid temperature (C)
+   * @return derivative of CO2 viscosity wrt density
+   */
+  Real dSupercriticalViscosity_dDensity(Real pressure, Real temperature) const;
+
+  /**
+   * Derivative of the viscosity of supercritical CO2 as a function of pressure
+   *
+   * @param pressure fluid pressure (Pa)
+   * @param temperature fluid temperature (C)
+   * @return derivative of CO2 viscosity wrt pressure
+   */
+  Real dSupercriticalViscosity_dP(Real pressure, Real temperature) const;
+
+  /**
+   * Henry's law constant coefficients for dissolution of CO2 into water.
+   * From Guidelines on the Henry's constant and vapour
+   * liquid distribution constant for gases in H20 and D20 at high
+   * temperatures, IAPWS (2004).
+   *
+   * @return constants for Henry's constant (-)
+   */
+  std::vector<Real> henryConstants() const;
```

Comment:
Henry's constant dictate how much of one species (CO2) is dissolved in a particular phase (water), and as such they affect the density and viscosity of the phase.  At present i think we haven't any tests, kernels or materials that actually use these things, but in the near future (within a month i hope) we will be using them a lot.